leader/leader.go

```diff
@@ -291,6 +292,10 @@ func getNode(ctx context.Context, client crclient.Client, nodeName string, node
 		log.Error(err, "Failed to get Node", "Node.Name", nodeName)
 		return err
 	}
+	// .Get() clears the APIVersion and Kind,
+	// so we need to set them before returning the object.
+	node.TypeMeta.APIVersion = "v1"
```

Comment:
Having to mess with TypeMeta manually is a bit of a smell - why did we need this?

Reply:
Did this for precedence and a test that used the TypeMeta information: https://github.com/operator-framework/operator-lib/blob/3bfabe03dfae8754ef3a3dc02b09ec618e2ecb78/leader/leader.go#L277-L280